src/twisted/names/test/test_names.py

```diff
@@ -1221,13 +1223,50 @@ def loadBindString(self, s):
         @return: a new bind authority
         @rtype: L{twisted.names.authority.BindAuthority}
         """
-        fp = FilePath(self.mktemp().encode("ascii"))
+        fp: FilePath[AnyStr] = FilePath(path)
         fp.setContent(s)
 
         return authority.BindAuthority(fp.path)
 
     def setUp(self):
-        self.auth = self.loadBindString(sampleBindZone)
+        path = self.mktemp().encode("ascii")
```

Comment:
Not all filesystems are ASCII, so this may fail randomly.  I think you want `os.fsencode` here if you really need to encode a path to bytes (or maybe just `FilePath(self.mktemp).asBytesMode().path`